src/cocotb/handle.py

```diff
@@ -151,26 +164,54 @@ def __repr__(self) -> str:
         return type(self).__qualname__ + "(" + desc + ")"
 
 
-IndexType = TypeVar("IndexType")
+KeyType = TypeVar("KeyType")
 
 
-class RegionObject(SimHandleBase, Generic[IndexType]):
-    """A region object, such as a scope or namespace.
+class HierarchyObjectBase(SimHandleBase, Generic[KeyType]):
+    """Base class for hierarchical objects.
 
-    Region objects don't have values, they are effectively scopes or namespaces.
+    Hierarchical objects don't have values, they are just scopes/namespaces of other objects.
+    This includes array-like hierarchical structures like "generate loops"
+    and named hierarchical structures like "generate blocks" or "module"/"entity" instantiations.
+
+    This base class defines logic to discover, cache, and inspect child objects.
+    It provides a :class:`dict`-like interface for doing so.
+
+    :meth:`_keys`, :meth:`_values`, and :meth:`_items` mimic their :class:`dict` counterparts.
+    One can also iterate over an object, which returns child objects, not keys;
```

Comment:
```suggestion
    You can also iterate over an object, which returns child objects, not keys;
```
We use "you" elsewhere already.